rfs_test/TEST_accountservice_schema.py

```diff
@@ -0,0 +1,262 @@
+# File: rfs_check.py
+# Description: Redfish service conformance check tool. This module contains implemented assertions for
+#   SUT.These assertions are based on operational checks on Redfish Service to verify that it conforms
+#   to the normative statements from the Redfish specification.
+#   See assertions in redfish-assertions.xlxs for assertions summary
+#
+# Verified/operational Python revisions (Windows OS) :
+#       3.5.2
+#
+# Initial code released : 01/2018
+#   Robin Thekkadathu Ronson ~ Texas Tech University
+#####################################################################################################
+
+import urllib3
+import json
+from openpyxl import load_workbook
+from openpyxl.styles import PatternFill
+from time import gmtime, strftime
+
+accSerData = None
+
+# current spec followed for these assertions
+REDFISH_SPEC_VERSION = "Version 1.2.2"
+
+
+###################################################################################################
+# Name: Assertion_ACCO101(self, log) :Account Service                                            
+# Assertion text: 
+#The value of this property shall be a boolean indicating whether this service is enabled.  If 
+#this is set to false, the AccountService is disabled.  This means no users can be created, 
+#deleted or modified.  Any service attempting to access the Account Service, like the Session 
+#Service, will fail accessing.  Thus new sessions cannot be started with the service disabled 
+#(though established sessions may still continue operating).  Note: this does not affect Basic 
+#AUTH connections.
+###################################################################################################
+
+def Assertion_ACCO101(self, log):
+
+    log.AssertionID = 'ACCO101'
+    assertion_status =  log.PASS 
+    log.assertion_log('BEGIN_ASSERTION', None)
+
+    relative_uris = self.relative_uris
+    authorization = 'on' 
+    rq_headers = self.request_headers()
+
+    json_payload, headers, status = self.http_GET('/redfish/v1/AccountService', rq_headers, authorization)
+
+    try:
+        isEnabled = json_payload['ServiceEnabled']
+    except:
+        assertion_status = log.WARN
+        log.assertion_log('line', "~ \'AccountsService\' not found in the payload from GET %s" % ('/redfish/v1/AccountService'))
+        return assertion_status
+    else:
+        try:
+            key = 'Accounts'
+            acc_collection = (json_payload[key])['@odata.id']
+        except:
+            assertion_status = log.WARN
+            log.assertion_log('line', "~ \'Accounts\' not found in the payload from GET %s" % ('/redfish/v1/AccountService'))
+            return assertion_status
+        else:
+            if not isEnabled:
+                rq_body = {'Name': 'Test'}
+                
+                json_payload, headers, status = self.http_PUT(acc_collection + '/test', rq_headers, rq_body, authorization)
```

Comment:
The user "test" may not be guaranteed to be there on a given system. It's probably safer to perform this check by performing a POST to the account collection and define an entirely new user. That operation should still fail if the account service is disabled.